sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -83,18 +85,37 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
         spanOptions: webResource.spanOptions
       }
     });
-    webResource.headers.set(Constants.HeaderConstants.AUTHORIZATION, `Bearer ${token}`);
+    if (token) {
+      webResource.headers.set(Constants.HeaderConstants.AUTHORIZATION, `Bearer ${token}`);
+    }
```

Comment:
This reminds me to a feedback you (@xirzec) gave me earlier. Why do we set the bearer token if we don't have a token? This seems like an improvement.